cudaextender/include/claraparabricks/genomeworks/cudaextender/utils.hpp

```diff
@@ -0,0 +1,150 @@
+///        target_position_in_read_2, query_position_in_read_2
+///        target_position_in_read_n, query_position_in_read_n
+///
+/// \param[out] seed_pairs Reference to vector into which parsed seed pairs are saved
+/// \param[in]  filepath   Reference to the string containing the path of the seed pairs csv
+inline void parse_seed_pairs(const std::string& filepath, std::vector<SeedPair>& seed_pairs)
+{
+    std::ifstream seed_pairs_file(filepath);
+    if (!seed_pairs_file.is_open())
+        throw std::runtime_error("Cannot open file");
+    if (seed_pairs_file.good())
+    {
+        std::string line;
+        while (std::getline(seed_pairs_file, line, ','))
+        {
+            SeedPair seed_pair;
+            seed_pair.target_position_in_read = std::atoi(line.c_str());
+            std::getline(seed_pairs_file, line); // Get the next value
+            seed_pair.query_position_in_read = std::atoi(line.c_str());
+            seed_pairs.push_back(seed_pair);
+        }
+    }
+}
+
+/// \brief Parses scored segment pairs from a csv file in the following format:
+///        target_position_in_read_1, query_position_in_read_1, length_1, score_1
+///        target_position_in_read_2, query_position_in_read_2, length_2, score_2
+///        target_position_in_read_n, query_position_in_read_n, length_n, score_n
+///
+/// \param[out] scored_segment_pairs Reference to vector into which parsed scored segment pairs are saved
+/// \param[in]  filepath   Reference to the string containing the path of the scored segment pairs csv
+inline void parse_scored_segment_pairs(const std::string& filepath, std::vector<ScoredSegmentPair>& scored_segment_pairs)
+{
+    std::ifstream scored_segment_pairs_file(filepath);
+    if (!scored_segment_pairs_file.is_open())
+        throw std::runtime_error("Cannot open file");
+    if (scored_segment_pairs_file.good())
+    {
+        std::string line;
+        while (std::getline(scored_segment_pairs_file, line, ','))
+        {
+            ScoredSegmentPair scored_segment_pair;
+            scored_segment_pair.seed_pair.target_position_in_read = std::atoi(line.c_str());
+            std::getline(scored_segment_pairs_file, line, ',');
+            scored_segment_pair.seed_pair.query_position_in_read = std::atoi(line.c_str());
+            std::getline(scored_segment_pairs_file, line, ',');
+            scored_segment_pair.length = std::atoi(line.c_str());
+            std::getline(scored_segment_pairs_file, line);
+            scored_segment_pair.score = std::atoi(line.c_str());
+            scored_segment_pairs.push_back(scored_segment_pair);
+        }
+    }
+}
+
+/// \brief Encodes character sequence as integer sequence
+///
+/// \param[out] dst_seq    Pointer to pre-allocated storage for encoded sequence
+/// \param[in]  src_seq    Pointer to input sequence
+/// \param[in]  length     Length of the sequence
+inline void encode_sequence(int8_t* dst_seq, const char* src_seq, const int32_t& length)
```

Comment:
Please change `const int32_t&` to `const int32_t`.

Reply:
Done.